pkg/3scale/amp/component/apicast_options.go

```diff
@@ -16,6 +16,8 @@ type ApicastOptions struct {
 	StagingResourceRequirements    v1.ResourceRequirements `validate:"-"`
 	ProductionReplicas             int32
 	StagingReplicas                int32
+	StagingLabels                  map[string]string `validate:"required"`
```

Comment:
The labels `app`, `threescale_component`, `threescale_component_element` are used on different types of K8s objects. You can see them in Services, DeploymentConfigs(don't confuse it with PodTemplates in the DCs), ConfigMaps, PDBs.

Maybe we should name this options as StagingPodTemplateLabels and ProductionPodTemplateLabels?

Additionally, is there some way we can reuse the values of labels for the other K8s objects? If you look at the pkg/3scale/amp/component/apicast.go you'll see the labels are mentioned are found in other objects too.

Reply:
done